crates/ty_python_semantic/src/types.rs

```diff
@@ -1315,6 +1335,26 @@ impl<'db> Type<'db> {
                 self_typevar == other_typevar
```

Comment:
This condition is also removable because it's handled in gradual equivalence, which is checked above.